File: src/ShortestWordDistance.java
class ShortestWordDistance {
    public int shortestDistance(String[] arr, String word1, String word2) {

        int word1Match = -1;
        int word2Match = -1;
        int minDistance = Integer.MAX_VALUE;

        for (int i = 0; i < arr.length; i++) {
            if (arr[i].equals(word1)) {
                word1Match = i;
                if (word2Match != -1) {
                    minDistance = Math.min(minDistance, Math.abs(word1Match - word2Match));
                }
            } else if (arr[i].equals(word2)) {
                word2Match = i;
                if (word1Match != -1) {
                    minDistance = Math.min(minDistance, Math.abs(word1Match - word2Match));
                }
            }
        }
        return minDistance;
    }
}

File: src/ShortestWordDistance.kt
class ShortestWordDistance {
    fun shortestDistance(wordsDict: Array<String>, word1: String, word2: String): Int {
        var minDistance = Int.MAX_VALUE
        var firstWord: Int? = null
        var secondWord: Int? = null

        for (i in 0..wordsDict.size - 1) {
            if (wordsDict[i] == word1) {
                firstWord = i
                if (secondWord != null) {
                    minDistance = Math.min(minDistance, firstWord - secondWord)
                }
            } else if (wordsDict[i] == word2) {
                secondWord = i
                if (firstWord != null) {
                    minDistance = Math.min(minDistance, secondWord - firstWord)
                }
            }
        }

        return minDistance
    }
}